internal/engine/synclet_build_and_deployer.go

```diff
@@ -134,7 +136,9 @@ func (sbd *SyncletBuildAndDeployer) getContainerForBuild(ctx context.Context, bu
 	defer span.Finish()
 
 	// get pod running the image we just deployed
-	pID, err := sbd.kCli.PodWithImage(ctx, build.Image)
+	// TODO(maia): parallelize this polling (inefficient b/c first we deploy all manifests in series,
+	// then we poll for pods for all of them (again in series)
+	pID, err := sbd.kCli.PollForPodWithImage(ctx, build.Image, time.Second*45)
```

Comment:
i remain skeptical of any api that blocks the current goroutine...45 seconds isn't great.

there are lots of "normal" user stories where the pod might never come up, like if they made a mistake in the yaml, or if the server exits immediately due to a bug in their code